src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/RuleManagerImpl.java

```diff
@@ -511,6 +511,11 @@ private List<SpeakerData> buildSharedEndpointYFlowCommands(List<FlowPath> flowPa
         }
         MeterId sharedMeterId = yFlow.getSharedEndpointMeterId();
 
+        if (sharedMeterId == null) {
+            log.debug("Skip commands for the y-flow {} as the sharedMeterId is null", yFlow.getYFlowId());
+            return emptyList();
```

Comment:
Are there tests covering these new cases? Require non-null below might throw an exception and now that iteration is skipped, so it's a different behavior

Reply:
Line 549 contains code with generator. This generator generates OpenFlow commands. In out case it's meter and rule commands. If we do not have meterId then we should not generate meter commands, but we still must generate rules commands.

You code will successfully create Y flow withput meter, but this flow wouldn't transmit traffic. To check traffic transmission you can one of the following:
1. Create functional tests with traffic generations.
2. Generate traffic manually and look at rules counters and statistics in OpenTSDB. Example: https://github.com/telstra/open-kilda/issues/5017

Reply:
I was reading the code and thought it was only installing meter commands. In fact, I tried with the ping tool from the API and it works. I will try the traffic transmissions that you propose. Thanks

Reply:
Ping tool has one disadvantage: during ping it doesn't use ingress (first) and egress (last) rules in flow path. It tests only transit rules because of technical restrictions. As you removed ingress rules - ping was successful